packages/tabs/addon/components/nucleus-tabs/tab-list-item.js

```diff
@@ -0,0 +1,320 @@
+  * @public
+  *
+  */
+  init() {
+    super.init(...arguments);
+  }
+
+  /**
+  * didInsertElement
+  *
+  * @method didInsertElement
+  * @description lifecycle event
+  * @public
+  *
+  */
+  didInsertElement() {
+    get(this, 'registerTabListItem').call(this, {
+      id: get(this, 'elementId'),
+      name: get(this, 'name')
+    });
+  }
+
+  /**
+  * mouseDown
+  *
+  * @method mouseDown
+  * @description event handler : We are negating the style applied on click-focus
+  * @public
+  *
+  */
+  mouseDown() {
+    if(get(this, 'isDisabled') === false) set(this, 'isPressed', true);
+  }
+
+  /**
+  * focusOut
+  *
+  * @method focusOut
+  * @description event handler : Removing style that was applied during press to negate focus
+  * @public
+  *
+  */
+  focusOut() {
+    if(get(this, 'isPressed') === true) set(this, 'isPressed', false);
+  }
+
+  /**
+  * click
+  *
+  * @method click
+  * @description event handler
+  * @public
+  *
+  */
+  click(event) {
+    if(get(this, 'isDisabled') === false) {
+      get(this, 'handleActivateTab').call(this, get(this, 'name'), event);
+    }
+  }
+
+  /**
+  * keyDown
+  *
+  * @method keyDown
+  * @description event handler
+  * @public
+  *
+  */
+  keyDown(event) {
+    event.stopPropagation();
+    const targetElement = event.target;
+    const firstElement = targetElement.parentElement.firstElementChild;
+    const lastElement = targetElement.parentElement.lastElementChild;
+
+    const keyCode = TABS_KEY_CODE;
+    switch (event.keyCode) {
+      case keyCode.ENTER:
+      case keyCode.SPACE:
+        event.preventDefault();
+        targetElement.click();
+        break;
+      case keyCode.END: 
+        lastElement.focus();
+        break;
+      case keyCode.HOME: 
+        firstElement.focus();
+        break; 
+      case keyCode.LEFT: 
+        get(this, '_focusPreviousTab').call(this, targetElement);
+        break;
+      case keyCode.RIGHT: 
+        get(this, '_focusNextTab').call(this, targetElement);
+        break;
+      default:
+        break;
+    }
+  }
+
+  /**
+  * _focusNextTab
+  * When last item, focus must circle back to previous item.
+  *
+  * @method _focusNextTab
+  * @description focus the next tab that is not disabled
+  * @private
+  *
+  */
+  _focusNextTab(element, elementInFocus) {
+    const nextElement = (element.nextElementSibling)? element.nextElementSibling : element.parentElement.firstElementChild;
+    if(elementInFocus && (elementInFocus.id === nextElement.id)) {
+      return;
+    } else if(nextElement.getAttribute('tabindex') === null) {
+      get(this, '_focusNextTab').call(this, nextElement, element);
+    } else {
+      nextElement.focus();
+    }
+  }
+
+  /**
+  * _focusPreviousTab
+  * When first item, focus must go back to last item.
+  *
+  * @method _focusPreviousTab
+  * @description focus the previous tab that is not disabled
+  * @private
+  *
+  */
+  _focusPreviousTab(element, elementInFocus) {
```

Comment:
This function is almost the same as `_focusNextTab`.  So can you refactor this into a single function with an additional param to address both the use cases?

Reply:
Their primary functionality is the same, but they simply operate on different nodes. Combining them makes more mess than solve something.

Reply:
```js
_focusTab(element, elementInFocus, direction) {
    let siblingElement;
    if(direction) {
        siblingElement  = (element.nextElementSibling)? element.nextElementSibling : element.parentElement.firstElementChild;
    } else {
      siblingElement = (element.previousElementSibling)? element.previousElementSibling : element.parentElement.lastElementChild;
}
   
    if(elementInFocus && (elementInFocus.id === siblingElement.id)) {
      return;
    } else if(siblingElement.getAttribute('tabindex') === null) {
      get(this, '_focusTab').call(this, siblingElement, element, direction);
    } else {
      siblingElement.focus();
    }
  }
```

Reply:
> D.R.Y: Don't Repeat Yourself. 

Having two functions with almost the same lines of code is a prime candidate to apply this principle.

Reply:
Done.